compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -598,13 +607,19 @@ class ClassfileParser(
       case _ =>
 
     val nargs = in.nextChar
-    val argbuf = new ListBuffer[untpd.Tree]
+    val argbuf = new ListBuffer[untpd.Tree | (Context => untpd.Tree)]
     var hasError = false
     for (i <- 0 until nargs) {
       val name = pool.getName(in.nextChar)
       parseAnnotArg(skip) match {
-        case Some(arg) => argbuf += untpd.NamedArg(name.name, arg)
-        case None => hasError = !skip
+        case Some(tag: EnumTag) =>
+          argbuf += ((ctx: Context) => untpd.NamedArg(name.name, enumTagTree(tag)(using ctx)))
```

Comment:
Can we go further and store `(name, tag)` directly in `argbuf` and thus the `lazyBuf` of `ClassfileAnnotation`?

Reply:
We can do that, but we need to refactor a little bit `ClassfileAnnotation` to delay the creation of `NamedArg`.